postgresql/provider.go

```diff
@@ -76,6 +76,32 @@ func Provider() terraform.ResourceProvider {
 				Optional:   true,
 				Deprecated: "Rename PostgreSQL provider `ssl_mode` attribute to `sslmode`",
 			},
+			"clientcert": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "SSL client certificate if required by the database.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cert": {
+							Type:        schema.TypeString,
+							Description: "The SSL client certificate file path. The file must contain PEM encoded data.",
+							Required:    true,
+						},
+						"key": {
+							Type:        schema.TypeString,
+							Description: "The SSL client certificate private key file path. The file must contain PEM encoded data.",
+							Required:    true,
+						},
+						"rootcert": {
```

Comment:
I wonder if ``rootcert`` must be in this nested block with client cert/key?

If I understood correctly, the CA  (``rootcert``) is used to verify the server certificate (so authenticate the server).
The client cert is used to authenticate the client.

So you can specify only ``cert``/``key`` if you want to authenticate the client or only ``rootcert`` if you want to authenticate the server (and all of them if you want to authenticate both).

So wouldn't it be better to keep ``cert``  and ``key`` in this ``clientcert`` block  but have ``sslrootcert` outside of it?

Reply:
Sure, that would be totally fine with me.

Reply:
Ok, I've made the requested changes, thank you for your feedback!